packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -877,10 +877,14 @@ contract VotingV2 is
             if (!_priceRequestResolved(priceRequest, voteInstance, currentRoundId)) {
                 // If the request is not resolved and the lastVotingRound less than the current round then the vote
                 // must have been rolled. In this case, update the internal trackers for this vote.
-
                 if (priceRequest.lastVotingRound < currentRoundId) {
                     priceRequest.lastVotingRound = SafeCast.toUint32(currentRoundId);
-                    skippedRequestIndexes[requestIndex] = requestIndex;
+
+                    // This is a subtle operation. This is not setting the skip value for the _current request_ to this
+                    // value. It is setting the skip value for the element after the last processed index to the skip
+                    // value. This causes this skip interval to extend on each subsequent rolled request because no
+                    // new elements are processed on a skip, thereby leaving nextIndexToProcess the same.
+                    skippedRequestIndexes[nextIndexToProcess] = requestIndex;
                     priceRequest.priceRequestIndex = SafeCast.toUint64(priceRequestIds.length);
                     priceRequestIds.push(priceRequestIds[requestIndex]);
```

Comment:
Note: we no longer need the if(...) indexTo++.

This was possible because the bottom if statement no longer uses indexTo, so there's no requirement that we iterate past the initial length since anything added mid-loop _must_ be a future round and therefore, unprocessable.